services/user/CommonApi/common/packages/plugin-tester/ui/src/components/editors/RawParameterEditor.tsx

```diff
@@ -0,0 +1,34 @@
+import { FC } from "react";
+
+interface RawParameterEditorProps {
+  values: Record<string, unknown>;
+  onChange: (values: Record<string, unknown>) => void;
+}
+
+export const RawParameterEditor: FC<RawParameterEditorProps> = ({
+  values,
+  onChange,
+}) => {
+  const handleChange = (newValue: string) => {
+    try {
+      const parsed = JSON.parse(newValue);
+      if (
+        typeof parsed === "object" &&
+        parsed !== null &&
+        !Array.isArray(parsed)
+      ) {
+        onChange(parsed as Record<string, unknown>);
+      }
+    } catch {
+      console.error("Invalid JSON");
+    }
+  };
+
+  return (
+    <textarea
+      className="common-textarea"
+      value={JSON.stringify(values, null, 2)}
+      onChange={(e) => handleChange(e.target.value)}
```

Comment:
Doing this `onChange` means the user really cannot free type in the field. Every key pressed must result in good JSON or the change doesn't happen. I assume that's what you're going for, right? (Makes sense, I guess. Keeps it valid and in sync with the rich editor.) Alternatively, you can validate on submit.